### PR TITLE
Add PR test to build kernels and upload artifacts

### PR DESCRIPTION
**Pull request by loadams**
- [X] Add .yml file that allows for automated build of DS-Kernels.
- [X] Add way to reduce parallel make jobs to use less memory to run in the runners provided.

### CODE REVIEW

File: builder/builder.py [hunk 1]
```diff
@@ -72,8 +72,12 @@ def build_extension(self, ext):
...
-        subprocess.check_call(['cmake', '-B', abs_build_temp, 
+        subprocess.check_call(['cmake', '-B', abs_build_temp,
                                f'-DLIB_OUTPUT_DIR={abs_build_lib}',
                                f'-DCUDA_ARCH_LIST={cuda_arch_list}'],
                                cwd=ext.source)
-        subprocess.check_call(['make', '-j'], cwd=abs_build_temp)
+
+        # Allow user to specify degree of make parallelism
+        make_jobs = os.environ.get('DS_KERNELS_MAKE_JOBS', None)
+        make_cmd = f"make -j {make_jobs}" if make_jobs else "make -j"
```
Comment (mrwyattii) [inline builder/builder.py:82]:
```suggestion
        make_cmd = f"make -j {make_jobs}" if make_jobs is not None else "make -j"
```

File: release/release.sh [hunk 2]
```diff
@@ -2,6 +2,8 @@ set -ex
 
 rm -rf dist
 
+#export DS_KERNELS_MAKE_JOBS=10
```
Comment (mrwyattii) [inline release/release.sh:5]: Any reason to keep this? Perhaps we can add a comment indicating users can uncomment this if they want to control the number?

**Review by mrwyattii — APPROVED** [review]

File: builder/builder.py (hunk 1 above)
Comment (loadams) [reply builder/builder.py]: Fixed now.

File: release/release.sh (hunk 2 above)
Comment (loadams) [reply release/release.sh:6]: Added a comment.

**Pull request merged**